tests/test_types.py

```diff
@@ -0,0 +1,135 @@
+r"""*Submodule for test suite of the* ``stdio_mgr.types`` components.
+
+**Author**
+    John Vandenberg (jayvdb@gmail.com)
+
+**File Created**
+    6 Sep 2019
+
+**Copyright**
+    \(c) Brian Skinn 2018-2019
+
+**Source Repository**
+    http://www.github.com/bskinn/stdio-mgr
+
+**Documentation**
+    See README.rst at the GitHub repository
+
+**License**
+    The MIT License; see |license_txt|_ for full license terms
+
+**Members**
+
+"""
+
+import io
+
+import pytest
+
+from stdio_mgr.stdio_mgr import _IMPORT_SYS_STREAMS, _RUNTIME_SYS_STREAMS
+from stdio_mgr.types import ClosingStdioTuple, FakeIOTuple, TextIOTuple
+
+
+def _create_dummy_wrapper():
+    return io.TextIOWrapper(io.BufferedRandom(io.BytesIO()))
+
+
+def _create_dummy_wrapper_tuple():
+    return (
+        io.TextIOWrapper(io.BufferedRandom(io.BytesIO())),
+        io.TextIOWrapper(io.BufferedRandom(io.BytesIO())),
+        io.TextIOWrapper(io.BufferedRandom(io.BytesIO())),
+    )
+
+
+def test_sys_module(request):
+    """Confirm sys module objects are of expected types."""
+    assert isinstance(_RUNTIME_SYS_STREAMS, TextIOTuple)
+
+    # This probably isnt enough, due to capture.py _py36_windowsconsoleio_workaround
+    if request.config.known_args_namespace.capture == "no":
+        assert isinstance(_RUNTIME_SYS_STREAMS, TextIOTuple)
+    else:
+        assert isinstance(_IMPORT_SYS_STREAMS, FakeIOTuple)
+
+
+def test_stdio_base_close():
+    """Test StdioTupleBase close()."""
+    cm = ClosingStdioTuple((io.BytesIO(), io.BytesIO(), io.BytesIO()))
+
+    with cm:
+        assert not cm.stdin.closed
+        assert not cm.stdout.closed
+        assert not cm.stderr.closed
+        assert not cm._any("closed")
+
+    assert cm.stdin.closed
+    assert cm.stdout.closed
+    assert cm.stderr.closed
+    assert cm._all("closed")
+
+    cm = ClosingStdioTuple(_create_dummy_wrapper_tuple())
+
+    # Detaching works ok.
+    cm.stdout.detach()
+
+    cm = ClosingStdioTuple(_create_dummy_wrapper_tuple())
+
+    # Detaching causes the __exit__ to fail
+    with pytest.raises(ValueError) as err:
+        with cm:
+            cm.stdout.detach()
+
+    assert str(err.value) == "underlying buffer has been detached"
+
+
+class _SafeCloseIOTuple(ClosingStdioTuple):
```

Comment:
Why is this defined *de novo* here? Is it just temporary, until it becomes sensible to move it to the actual codebase?